### DISCUSSION TITLE
Can button have this option?

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#4001] [Q&A]
## The requirement is to be able to change the size

##The following is the effect

![截屏2023-06-25 13 11 54](https://github.com/fyne-io/fyne/assets/1819457/b9b5a33a-3117-4888-9a2c-9260e4ae1724)

**Comment by andydotxyz** [comment]
That looks like internal padding, it is controlled by the theme - a custom theme could override it as you illustrate.

**Comment by junmaqiang** [comment]
I hope the official can make it a configurable item
 
widget.LargeImportance 
widget.DefaultImportance 
widget.SmallImportance

**Reply by andydotxyz** [reply]
No, all the standard widgets have a standard size - it's so the alignment works out by default.

Importance is shown by colour - this allows the layout to continue to flow.

**Reply by andydotxyz** [reply]
If you want to make a button that is bigger then use a container layout that gives it more space. If you want one that is smaller then you can override it and return a MinSize that is smaller than the standard minimum.

**Reply by junmaqiang** [reply]
If the user defines it casually, it will be messy. It is best to officially provide the standard and the user can refer to it.

## For example
`
widget.LargeSize

widget.DefaultSize

widget.SmallSize
`

**Reply by andydotxyz** [reply]
The theme, and toolkit, has a single standard size for widgets. Adding more will be messy and complicated.

Remember this is a semantic api - a "large button" does not have meaning like a "primary button" does. Likewise "heading text" does have meaning but "larger text" does not.

This is our design and we stick to it.

**Comment by junmaqiang** [comment]
this file "fyne.io/fyne/v2@v2.3.5/widget/button.go " can add ButtonIconTopText ?

```
const (
	// ButtonIconLeadingText aligns the icon on the leading edge of the text.
	ButtonIconLeadingText ButtonIconPlacement = iota
	// ButtonIconTrailingText aligns the icon on the trailing edge of the text.
	ButtonIconTrailingText

        //    
        ButtonIconTopText     // Can add this option?  
)

```

this demo

![截屏2023-06-28 21 19 25](https://github.com/fyne-io/fyne/assets/1819457/b6bad62b-4f19-46de-a3c2-a42d827b1025)

**Reply by matwachich** [reply]
I think this ks supported in ToolBar Buttons

**Reply by andydotxyz** [reply]
Remember that the Button widget is just a basic standard button, it promises to align and fit with all the other standard content. It is not difficult to build a vertically aligned button with a larger icon (like app tabs etc) but having that as a core feature of button will break layouts.
